insights/parsers/systemd/unitfiles.py

```diff
@@ -69,15 +56,99 @@ def parse_content(self, content):
                 self.services[service] = enabled
                 self.parsed_lines[service] = line
                 self.service_list.append(service)
+            elif any(part in valid_states for part in parts):
+                self.service_details['ACTIVE'] = parts[2]
+                self.service_details['SUB'] = parts[3]
+                if any(unit in parts[0] for unit in valid_units):
+                    self.service_details['UNIT'] = parts[0]
+                    self.unit_list[parts[0]] = self.service_details
+                self.service_details = {}
 
     def is_on(self, service_name):
         """
         Checks if the service is enabled in systemctl.
 
-        Args:
+        Args::
             service_name (str): service name including '.service'
 
-        Returns:
+        Returns::
             bool: True if service is enabled, False otherwise
         """
         return self.services.get(service_name, False)
+
+    def get_service_details(self, service_name):
+        """
+        Return the service details collected by systemctl.
+
+        Args::
+            service_name (str): service name including its extension.
+        Returns::
+            >> service_name = "neutron-dhcp-agent.service"
+            >> self.data.get(service_name)
+            {'ACTIVE': 'active', 'SUB': 'running', 'UNIT': 'neutron-dhcp-agent.service'}
+        """
+        return self.unit_list.get(service_name, False)
```

Comment:
The use case for this seems to be:
```
svc = conf.get_service_details('nonexistent.service')
if svc['ACTIVE'] == 'active':
```
Which will cause a KeyError if given a blank dict and AttributeError if given False.  Obviously we'd like to avoid bad usage (the user should use `is_on()`) but we'd also like to avoid breaking if the user is for some reason trying to get the details of a service that doesn't exist.

I guess my thought here is that this should really be a __get__ method so that the object can be used as a pseudo-dictionary for services.  This way it's more obvious that 'getting a service that doesn't exist' causes a KeyError.